storage/atsd/atsd.go

```diff
@@ -0,0 +1,168 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package atsd
+
+import (
+	"flag"
+
+	info "github.com/google/cadvisor/info/v1"
+	"github.com/google/cadvisor/manager"
+	"github.com/google/cadvisor/storage"
+	atsdstorage "github.com/google/cadvisor/storage/atsd/storage"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	startDelay           = 15 * time.Second
+	timestampPeriodError = 250 * time.Millisecond
+)
+
+var (
+	url        = flag.String("atsd_storage_url", "", "atsd http/https endpoint")
+	writeHost  = flag.String("atsd_storage_write_host", "", "atsd write host endpoint")
+	protocol   = flag.String("atsd_storage_write_protocol", "", "atsd write protocol (tcp/udp)")
+	dockerHost = flag.String("atsd_storage_docker_host", "docker-host", "hostname of docker machine")
```

Comment:
Docker host is the entity in ATSD. It is useful to set this flag as the machine hostname where docker daemon is running.